File: Domain/MovieRecognition.cs
namespace Domain;

public class MovieRecognition(Guid userId, Uri videoUrl)
{
	public Guid Id { get; init; } = Guid.NewGuid();
	public Guid UserId { get; } = userId;
	public Uri VideoUrl { get; } = videoUrl;
	public MovieRecognitionStatus Status { get; set; } = MovieRecognitionStatus.Created;
	public DateTime CreatedAt { get; init; } = DateTime.UtcNow;
	public Guid? VideoId { get; set; }
	public RecognizedTitle? RecognizedMovie { get; set; }
	public string? FailureMessage { get; set; }
	public bool? RecognizedCorrectly { get; set; }
}

public enum MovieRecognitionStatus
{
	Created,
	InProgress,
	Failed,
	Succeeded,
	Invalid
}

File: frontend/types/MovieRecognition.ts
import RecognizedTitle from './RecognizedTitle';
import Video from './Video';

interface MovieRecognition {
  id: string;
  user_id: string;
  video_url: string;
  created_at: string;
  status: MovieRecognitionStatus;
  failure_message?: string;
  video?: Video;
  recognized_movie?: RecognizedTitle;
  recognized_correctly?: boolean;
}

export type MovieRecognitionStatus = 'Created' | 'InProgress' | 'Failed' | 'Succeeded' | 'Invalid';

export default MovieRecognition;
